Realm/Realm/Native/PrimitiveValue.cs

```diff
@@ -28,10 +28,6 @@ namespace Realms.Native
     [DebuggerDisplay("PrimitiveValue({Type})")]
     internal unsafe struct PrimitiveValue
     {
-        [FieldOffset(0)]
-        [MarshalAs(UnmanagedType.U1)]
-        private bool bool_value;
```

Comment:
Removing the `[MarshalAs]` attribute from `bool_value` appears to fix the issue.